### PR TITLE
Implementing Simple Events and Alarms & Condition simulations

### PR DESCRIPTION
**Pull request by buzzfrog**
## Purpose

To have the possibility to simulate these features. The code is mainly copied from OPC Foundations sample repository.

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```
I hope not. I haven't tested it in combination with other features and in a container.

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
* Use command parameter `--ses´ to test Simple Events.
* Use command parameter `--alm to test Alarms & Condition.

## Other Information

As I said above I have mainly copied code from the OPC Foundation repository, including the headers with copyright information. I don't know if that can be any problems with that?

### CODE REVIEW

File: src/PlcNodeManager.cs [hunk 1]
```diff
@@ -186,7 +186,7 @@ private void AddSpecialNodes(FolderState dataFolder)
         {
             if (PlcSimulation.AddSpecialCharName)
             {
-                const string SpecialChars = "\"!�$%&/()=?`�\\+~*'#_-:.;,<>|@^���{[]}";
+                const string SpecialChars = "\"!�$%&/()=?`�\\+~*'#_-:.;,<>|@^���{[]}";
```
Comment (luiscantero) [inline src/PlcNodeManager.cs:189]: Pls revert this change

**Review by luiscantero — CHANGES_REQUESTED** [review]

File: src/PlcNodeManager.cs (hunk 1 above)
Comment (buzzfrog) [reply src/PlcNodeManager.cs:189]: When we are at it:
I always get an exception: _plcServer.PlcNodeManager.SpecialCharNameNode.Stop(); (133)
when I close the program.

File: src/AlarmCondition/Model/AreaConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+namespace AlarmCondition
+{
+    public class AreaConfiguration
+    {
+        public string Name { get; set; }
+        public AreaConfigurationCollection SubAreas { get; set; }
+        public StringCollection SourcePaths { get; set; }
+
+
```
[the remark this answers is not in the document]
Comment (buzzfrog) [reply src/AlarmCondition/Model/AreaConfiguration.cs]: done

File: src/PlcNodeManager.cs (hunk 1 above)
Comment (luiscantero) [reply src/PlcNodeManager.cs]: Could you do me a favor and change the following:
In PlcNodeManager to avoid this type of encoding issues:
string SpecialChars = HttpUtility.HtmlDecode(@"&quot;!&#167;$%&amp;/()=?`&#180;\+~*&#39;#_-:.;,&lt;&gt;|@^&#176;€&#181;{[]}");

In PlcSimulation, it should fix the Stop error:
_plcServer.PlcNodeManager.SpikeNode?.Stop();
_plcServer.PlcNodeManager.DipNode?.Stop();
_plcServer.PlcNodeManager.PosTrendNode?.Stop();
_plcServer.PlcNodeManager.NegTrendNode?.Stop();
_plcServer.PlcNodeManager.AlternatingBooleanNode?.Stop();
_plcServer.PlcNodeManager.StepUpNode?.Stop();
_plcServer.PlcNodeManager.RandomSignedInt32?.Stop();
_plcServer.PlcNodeManager.RandomUnsignedInt32?.Stop();
_plcServer.PlcNodeManager.SpecialCharNameNode?.Stop();

Thanks!

**Review by luiscantero — APPROVED** [review]

**Comment by eoursel** [comment]
Thanks guys ... I am interested by the Alarm simulation feature.  Something which is really needed. My point is to simulate a huge flow of alarms.

**Comment by buzzfrog** [comment]
@eoursel 
Our team, not this one, are in the envision phase to create a more advanced alarm simulator. At this moment I'm not sure if this will be included in this simulator.

File: README.md [hunk 3]
```diff
@@ -131,6 +131,14 @@ Features:
...
 
+## Simple Events
+
+The option `--ses` enables simple events [quickstart sample](https://github.com/OPCFoundation/UA-.NETStandard-Samples/tree/master/Workshop/SimpleEvents) from OPC Foundation.
+
+## Alarms and Condition
+
+The option `--alm` enables Alarm and Condition [quickstart sample](https://github.com/OPCFoundation/UA-.NETStandard-Samples/tree/master/Workshop/AlarmCondition) from OPC Foundation.
+
 ## Other features
```
Comment (hansgschossmann) [inline README.md:142]: Dag, can you add some description on which functionality is enabled with these options. We should at high level describe which events/alarms are generated and how/if they can be configured.

File: README.md [hunk 4]
```diff
@@ -131,6 +131,40 @@ Features:
 - When the heater is on, the bottom temperature increases by 1 degree/s, the top temperature is always 5 degrees less than the bottom one
 - Pressure is calculated as 100000 + bottom temperature
 
+## Simple Events
+
+The option `--ses` enables simple events [quickstart sample](https://github.com/OPCFoundation/UA-.NETStandard-Samples/tree/master/Workshop/SimpleEvents) from OPC Foundation.
```
Comment (luiscantero) [inline README.md:136]: Seems that there's a word missing here

Comment (buzzfrog) [reply README.md:136]: Is this enough? 
The option --ses enables simple events **from the** quickstart sample...

**Review by hansgschossmann — APPROVED** [review]
:shipit:
